### PR TITLE
payment details frontend & backend

### PR DESCRIPTION
**Pull request by WildlyManifest**
Backend needs a check.
Needs to be linked from profile page.

### CODE REVIEW

File: client/modules/PaymentDetails/PaymentDetails.js [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+  }
+
+  isValid(key) {
+    const valid = this.valids[key];
+    const touched = this.state.touched[key];
+    return valid || !touched;
+  }
+
+  formIsValid() {
+    return Object.keys(this.valids).some(field => this.valids[field] === false);
```
Comment (s3337852) [inline client/modules/PaymentDetails/PaymentDetails.js]: image
i tried a quick example of this validation in the dev tools console. its not returning the value you would expect. array.some() checks for existence of at least 1 element that matched the predicate, so if one field in valids == false, then it returns true

can we change it to Object.keys(this.valids).every(field => this.valids[field] === true)

Comment (s3337852) [reply client/modules/PaymentDetails/PaymentDetails.js]: or even change it to **!**Object.keys(this.valids).some(field => this.valids[field] === false);

File: server/controllers/paymentDetails/paymentDetails.controller.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        res.status(500).send(err);
+      } else if (!user) {
+        res.status(404).send();
+      } else {
+        res.status(200).send(user);
+      }
+    });
+};
+
+const changePaymentDetails = function(req, res) {
```
Comment (s3337852) [inline server/controllers/paymentDetails/paymentDetails.controller.js]: can we have 2 separate methods here, one for creation and one for updating
POST for creation
 -- find user by id
 -- create CreditCard
 -- save CreditCard
 -- set user.CreditCard
 -- save user
and PUT/:id for updating (the getMyPaymentDetails returns the whole credit card, so we have an id)
 -- CreditCard.findById
 -- update fields
 -- save changes

that way we don't have to do an inline deletion and its only 1 save

**Comment by WildlyManifest** [comment]
I've done the requested changes. 
Disabled the writing to database.
Link from Profile.

**Pull request merged**